lightning/src/util/persist.rs

```diff
@@ -243,10 +250,886 @@ where
+		read_only: bool,
+		entropy_source: ES,
+		signer_provider: SP,
+	) -> MonitorUpdatingPersister<Box<TestStore>, Box<TestLogger>, ES, SP>
+	where
+		ES::Target: EntropySource + Sized,
+		SP::Target: SignerProvider + Sized,
+	{
+		MonitorUpdatingPersister {
+			kv_store: Box::new(TestStore::new(read_only)),
+			logger: Box::new(TestLogger::new()),
+			maximum_pending_updates: 11,
+			entropy_source,
+			signer_provider,
+		}
+	}
+
+	#[test]
+	fn test_update_from_u64_and_as_str_works() {
+		let cases = [
+			(0, "00000000000000000000"),
+			(21, "00000000000000000021"),
+			(u64::MAX, "18446744073709551615"),
+		];
+		for (input, expected) in &cases {
+			let update_name = UpdateName::from(*input);
+			assert_eq!(update_name.as_str(), *expected);
+		}
+	}
+
+	#[test]
+	fn test_update_from_bad_string_fails() {
+		let cases = [
+			("deadbeef".to_string(), "cannot parse u64"),
+			("-0000000000000000001".to_string(), "cannot parse u64"),
+		];
+		for (input, expected) in &cases {
+			let got = UpdateName::new(input.to_owned())
+				.expect_err("bad update name accepted")
+				.to_string();
+			assert!(
+				got.contains(*expected),
+				"wrong update name error\nexpected: {}\ngot: {}",
+				expected,
+				got
+			);
+		}
+	}
+
+	#[test]
+	fn test_monitor_from_outpoint_and_as_str_works() {
+		let cases = [
+			(
+				OutPoint {
+					txid: Txid::from_hex(
+						"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef",
+					)
+					.unwrap(),
+					index: 1,
+				},
+				"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef_1",
+			),
+			(
+				OutPoint {
+					txid: Txid::from_hex(
+						"f33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeef",
+					)
+					.unwrap(),
+					index: u16::MAX,
+				},
+				"f33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeef_65535",
+			),
+		];
+		for (input, expected) in &cases {
+			let monitor_name = MonitorName::from(*input);
+			assert_eq!(monitor_name.as_str(), *expected);
+		}
+	}
+
+	#[test]
+	fn test_monitor_from_bad_string_fails() {
+		let cases = [
+			(
+				"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef".to_string(),
+				"No tx index",
+			),
+			(
+				"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef_65536"
+					.to_string(),
+				"Invalid tx index",
+			),
+			(
+				"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef_21".to_string(),
+				"Invalid tx ID",
+			),
+		];
+		for (input, expected) in &cases {
+			let got = MonitorName::new(input.to_owned())
+				.expect_err("bad monitor name accepted")
+				.to_string();
+			assert!(
+				got.contains(*expected),
+				"wrong monitor name error\nexpected: {}\ngot: {}",
+				expected,
+				got
+			);
+		}
+	}
+
+	#[test]
+	fn test_persister_with_real_monitors() {
+		let chanmon_cfgs = create_chanmon_cfgs(4);
+		// Easiest way to get the drop order right is to make the persisters first, then mutate them.
+		let throwaway_cfgs = create_node_cfgs(2, &chanmon_cfgs);
```

Comment:
You shouldn't need this if you use `&chanmon_cfgs[0].keys_manager` when initializing the persister.

Reply:
nice, that works. I think the exemplar I worked from used the node_cfg keys manager, and I didn't realize there was one in here I could use.